dawn/test/unit-test/dawn/Optimizer/Passes/TestComputeMaxExtent.cpp

```diff
@@ -47,18 +47,18 @@ class ComputeMaxExtents : public ::testing::Test {
     std::shared_ptr<SIR> sir =
         SIRSerializer::deserializeFromString(jsonstr, SIRSerializer::Format::Json);
 
-    std::unique_ptr<OptimizerContext> optimizer = compiler_.runOptimizer(sir);
+    auto stencilInstantiationMap = compiler_.optimize(compiler_.parallelize(sir));
```

Comment:
shouldn't this be the compile step? Can we be explicit?

Reply:
The problem is that we need the IIR for asserts below it, and compile only returns a translation unit.

Reply:
understood

Reply:
so again, we assume that `compile` is the full driver that goes from `SIR` to `TranslationUnit` and the individual steps `parallelize`, `optimize` and `generate` can be called independently?

Reply:
I replaced the chain of calls with a single overloaded `optimize` method. I agree that that makes the most sense in the current framework.